src/app/(Authentication)/SignUp/Address/index.tsx

```diff
@@ -0,0 +1,155 @@
+
+  const onChangeStreetAddress = (text: string) => {
+    setErrorExists(false);
+    setStreetAddress(text);
+  };
+
+  const onChangeCity = (text: string) => {
+    setErrorExists(false);
+    setCity(text);
+  };
+
+  const onChangeState = (text: string) => {
+    setErrorExists(false);
+    setState(text);
+  };
+
+  const onChangeZipcode = (text: string) => {
+    setErrorExists(false);
+    setZipcode(text);
+  };
+
+  const validateAddressInputs = () => {
+    // Insert any address input validation logic
+    return true;
+  };
+
+  const handleSubmit = () => {
+    if (validateAddressInputs()) {
+      signUpUser(name, email, password, streetAddress, city, state, zipcode);
+    }
+  };
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.header}>
+        <TouchableOpacity
+          style={styles.backButton}
+          onPress={() => router.back()}
+        >
+          <Text style={styles.backText}>Back</Text>
+        </TouchableOpacity>
+
+        <View style={styles.image}>
+          <Image
+            source={require('../../../../../assets/inline-logo.jpeg')}
+            style={{ width: 100, height: 12.5 }}
+          />
+        </View>
+      </View>
+
+      <Text style={styles.instructionText}>Last, enter your address.</Text>
+
+      <View style={styles.inputBox}>
+        <AuthInput
+          input={streetAddress}
+          onChangeInput={onChangeStreetAddress}
+          labelText="Street address"
+          placeholderText="Street address"
+          isPassword={false}
+          keyboard="default"
+          autoCapitalization
+        />
+      </View>
+
+      <View style={styles.inputBox}>
+        <AuthInput
+          input={city}
+          onChangeInput={onChangeCity}
+          labelText="City"
+          placeholderText="City"
+          isPassword={false}
+          keyboard="default"
+          autoCapitalization
+        />
+      </View>
+
+      <View style={styles.inputWrap}>
+        <View style={styles.smallInputBox}>
+          <AuthInput
+            input={state}
+            onChangeInput={onChangeState}
+            labelText="State"
+            placeholderText="State"
+            isPassword={false}
+            keyboard="default"
+            autoCapitalization
+            isHalfWidth
+          />
+        </View>
+
+        <Text style={styles.space}> </Text>
+
+        <View style={styles.smallInputBox}>
+          <AuthInput
+            input={zipcode}
+            onChangeInput={onChangeZipcode}
+            labelText="Zipcode"
+            placeholderText="Zipcode"
+            isPassword={false}
+            keyboard="default"
+            autoCapitalization
+            isHalfWidth
+          />
+        </View>
+      </View>
+      <Text style={styles.space2}> </Text>
+
+      <TouchableOpacity
+        disabled={
+          streetAddress.trim() === '' ||
+          city.trim() === '' ||
+          state.trim() === '' ||
+          zipcode.trim() === ''
```

Comment:
I tried using the `.trim()` method to prevent inputs only containing whitespace from passing, but idk if it works. Could you find a fix for this and make sure it works on other screens with similar non-empty input validation?

Reply:
When I test out this method it seems to work as inputs with only whitespace aren't accepted. This is also set on the password screen though, but I think it's fine when the password's comprised of spaces. Also when you enter spaces in the password secureInput it shows up as dots, so the discrepancy of not accepting it seems confusing. Unless u have some objections I'm going to remove it from that screen specifically